### PR TITLE
Add toSortedSet operation to BitSet

### PR DESCRIPTION
**Pull request by julienrf**
Fixes scala/collection-strawman#405

### CODE REVIEW

File: collections/src/main/scala/strawman/collection/BitSet.scala [hunk 1]
```diff
@@ -156,6 +157,9 @@ trait BitSetOps[+C <: BitSet with BitSetOps[C]]
 
   @`inline` final def ^ (other: BitSet): C = xor(other)
 
+  /** Upcasts this `BitSet` to a `SortedSet[Int]` */
+  def toSortedSet: SortedSet[Int]
```
Comment (julienrf) [inline collections/src/main/scala/strawman/collection/BitSet.scala]: I’ve added a `toSortedSet` method to `BitSet`.

File: collections/src/main/scala/strawman/collection/immutable/BitSet.scala [hunk 2]
```diff
@@ -49,6 +49,9 @@ sealed abstract class BitSet
   /** Update word at index `idx`; enlarge set if `idx` outside range of set.
     */
   protected def updateWord(idx: Int, w: Long): BitSet
+
+  override def toSortedSet: SortedSet[Int] = this
```
Comment (julienrf) [inline collections/src/main/scala/strawman/collection/immutable/BitSet.scala]: I override it in `immutable.BitSet` to return an `immutable.SortedSet[Int]`

File: collections/src/main/scala/strawman/collection/mutable/BitSet.scala [hunk 3]
```diff
@@ -138,6 +136,9 @@ class BitSet(protected[collection] final var elems: Array[Long])
     new BitSet(java.util.Arrays.copyOf(elems, elems.length))
 
   def toImmutable: immutable.BitSet = immutable.BitSet.fromBitMask(elems)
+
+  override def toSortedSet: SortedSet[Int] = this
```
Comment (julienrf) [inline collections/src/main/scala/strawman/collection/mutable/BitSet.scala]: Same override in `mutable.BitSet`, but which returns a `mutable.SortedSet[Int]`

File: collections/src/main/scala/strawman/collection/mutable/BitSet.scala [hunk 4]
```diff
@@ -85,8 +85,6 @@ class BitSet(protected[collection] final var elems: Array[Long])
 
   def get(elem: Int): Option[Int] = if (contains(elem)) Some(elem) else None
 
-  def unconstrained: collection.Set[Int] = this
```
Comment (julienrf) [inline collections/src/main/scala/strawman/collection/mutable/BitSet.scala]: I removed this method that doesn’t exist in the old collection and for which we already have `unsorted`.

**Comment by szeiger** [comment]
Requiring an explicit `toSortedSet` call seems wrong. I think the correct solution for #405 is to get the priorities of the overloads right so you don't need this method. Making `toSortedSet` an upcast is also inconsistent with other `to...` methods where we often refine the return types further (i.e. `BitSet.toSortedSet` could be expected to return a `BitSet`).

**Pull request closed without merging**